modules/Bio/EnsEMBL/DBSQL/SliceAdaptor.pm

```diff
@@ -2727,5 +2666,186 @@ sub _build_circular_slice_cache {
   $sth->finish();
 } ## end _build_circular_slice_cache
 
+=head2 fetch_by_seq_region_synonym
+
+  Args       : $cs, $coord_system_name, $seq_region_name, $start, $end, $strand, $version, $no_fuzz
+  Example    : $self->fetch_by_seq_region_synonym( $cs, $cs->name(), $seq_region_name, $start, $end, $strand, $version, $no_fuzz );
+  Description: fetches all the seq region synonyms (or uses wildcard) when requested
+  Returntype : Bio::EnsEMBL::SliceAdaptor, or none 
+  Exceptions : None
+  Caller     : general
+  Status     : (refactored from fetch_by_region)
+
+=cut
+
+sub fetch_by_seq_region_synonym {
+
+  my ( $self, $cs, $coord_system_name, $seq_region_name, $start, $end, $strand, $version, $no_fuzz ) = @_;
```

Comment:
I think this can be simplified to remove `$coord_system_name`. At this point, if you have a CoordSystem object, then you can get the name directly from it (with `$cs->name()`). The user-supplied `$coord_system_name` is not used anymore in this method, and we're past the point where we are checking for the special name "toplevel"

Reply:
Thanks @ens-bwalts , I've removed `$coord_system_name` from the input parameters to `fetch_by_seq_region_synonym` and modified the subroutine accordingly.